### DISCUSSION TITLE
If I want to modify the native code

### DISCUSSION THREAD

**Discussion by Chenjiujiu** [expo/expo#30033] [Archived: Questions]
If I want to modify the native code，exc： modify ios appDelegate.file or android app/bulid.gradle
But I hope to continue using CNG. What should I do

**Comment by lukmccall** [comment]
Hi @Chenjiujiu, we created the [config plugins](https://docs.expo.dev/config-plugins/introduction/) to solve that problem. The best place to start on how to make your own config plugin is our documentation — https://docs.expo.dev/config-plugins/plugins-and-mods/. Also, you can find some examples in the [`expo/expo`](https://github.com/expo/expo) repository. If you have more questions, let me know.
